### ISSUE TITLE
support  "extra_hosts" key-value to use  /etc/hosts file 

### ISSUE THREAD

**Issue by reynaldliu** [containerd/nerdctl#314]
You can add to nerdctl compose xxx.yml
Does the "extra_hosts" key-value pair support so that the container can be injected, added, modified, and the /etc/hosts file

**Comment by fahedouch** [comment]
resolved in https://github.com/containerd/nerdctl/pull/328

**Comment by polRk** [comment]
> resolved in [#328](https://github.com/containerd/nerdctl/pull/328)

It seems not:

```
  extra_hosts:
    - "ydb:172.28.0.10"
```

```
[2026-01-13 11:04:02] Bootstrapping cluster with endpoint: grpc://ydb:2136
Status: TRANSPORT_UNAVAILABLE
Issues:
<main>: Error: GRpc error: (14): connections to all backends failing; last error: UNKNOWN: ipv4:172.28.0.10:2136: Failed to connect to remote host: Connection refused
```


```
time="2026-01-13T14:08:50+03:00" level=fatal msg="failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running createRuntime hook #0: exit status 1, stdout: , stderr: time=\"2026-01-13T14:08:50+03:00\" level=warning msg=\"Container failed starting. Removing allocated network configuration.\"\ntime=\"2026-01-13T14:08:50+03:00\" level=fatal msg=\"failed to call cni.Setup: plugin type=\\\"bridge\\\" failed (add): failed to allocate for range 0: requested IP address 172.28.0.12 is not available in range set 172.28.0.1-172.28.255.254\""
```
